### PR TITLE
Tcp networking rework

### PR DESCRIPTION
**Pull request by Shosoul**
`tcp-nodelay` disables the Nagle algorithm.

### CODE REVIEW

**Comment by Shosoul** [comment]
I think it is in our best interest for now to keep our application-layer version of `keep-alive`, mainly the `ELOL` and `FLOL` response.

File: network.lua [hunk 1]
```diff
@@ -222,6 +222,7 @@ function network_init(ip, network_port)
     --error(loc("nt_conn_timeout"))
     return false
   end
+  logger.info(TCP_sock:setoption("tcp-nodelay", true) and "tcp-nodelay enabled" or "tcp-nodelay option failure")
```
Comment (JamesVanBoxtel) [inline network.lua:225]: Can we make each of the calls be on a separate line and save the result in a variable so they don't look like just log calls?

File: server.lua [hunk 2]
```diff
@@ -1577,8 +1578,11 @@ logger.info("initialized!")
 local prev_now = time()
 while true do
   server_socket:settimeout(0)
+  server_socket:setoption("tcp-nodelay", true)
   local new_conn = server_socket:accept()
   if new_conn then
+    new_conn:settimeout(0)
+    new_conn:setoption("tcp-nodelay", true)
```
Comment (JamesVanBoxtel) [inline server.lua:1585]: maybe capture this result too? and maybe "debug" is a better log level?

File: server.lua [hunk 3]
```diff
@@ -1525,6 +1524,8 @@ end
 --]]
 local server_socket = socket.bind("*", SERVER_PORT or 49569) --for official server
 --local server_socket = socket.bind("*", 59569) --for beta server
+server_socket:settimeout(0)
```
Comment (JamesVanBoxtel) [inline server.lua:1527]: do you know why we are calling settimeout with a value of 0? I can't find any documentation on what happens with a zero value

**Comment by JamesVanBoxtel** [comment]
Please also put in the final squash merge the justification and testing we have done to prove this reduces latency.

File: server_globals.lua [hunk 4]
```diff
@@ -30,6 +30,9 @@ PLACEMENT_MATCH_K = 50
 NAME_LENGTH_LIMIT = 16
 PLACEMENT_MATCHES_ENABLED = false
 COMPRESS_REPLAYS_ENABLED = false
+COMPRESS_SPECTATOR_REPLAYS_ENABLED = false -- Send current replay inputs over the internet in a compressed format to spectators who join.
+TCP_NODELAY_ENABLED = false -- Disables Nagle's Algorithm for TCP. Decreases data packet delivery delay, but increases amount of bandwidth and data used.
```
Comment (JamesVanBoxtel) [inline server_globals.lua:34]: shouldn't TCP_NODELAY_ENABLED be enabled by default since its the better option and the whole point of this PR?

Comment (Shosoul) [reply server_globals.lua:34]: Yea, I don't remember why I turned it off by default
```suggestion
TCP_NODELAY_ENABLED = true -- Disables Nagle's Algorithm for TCP. Decreases data packet delivery delay, but increases amount of bandwidth and data used.
```

**Review by JamesVanBoxtel — APPROVED** [review]

**Pull request merged**